### ISSUE TITLE
Rename logits to `model_output`

### ISSUE THREAD

**Issue by MischaPanch** [aai-institute/tianshou#917]
The TODO in the forward of `PGPolicy`  about renaming `logits` should be addressed. It will likely affect the naming in `ModelOutputBatchProtocol` and in other places, as well as docstrings. The exact usage of the model outputs should be clarified everywhere (they are passed to a torch `Distribution` which is then used for sampling or for computing `log_prob`).

_Originally posted by @Trinkle23897 in https://github.com/thu-ml/tianshou/pull/912#discussion_r1300865261_

**Comment by arnaujc91** [comment]
Will take a look at this one.

**Comment by MischaPanch** [comment]
We can't address all issues related to this, but the batch of improvements that can be done in version 1.x.y will be done in #1032
